indexer.ts

```diff
@@ -0,0 +1,743 @@
+        potUpdate["total_matching_pool_usd"] = pot.total_matching_pool_usd || 0 + total_amount_usd
+        potUpdate["total_matching_pool"] = pot.total_matching_pool || 0 + total_amount
+        potUpdate["matching_pool_donations_count"] = pot.matching_pool_donations_count || 0 + 1
+        let accountUpdate = {
+
+        }
+      } else {
+        potUpdate["public_donations_count"] = pot.public_donations_count || 0 + 1
+      }
+      await context.db.Pot.update({id: pot.id}, potUpdate)
+    }
+
+    let recipient = donation_data.project_id || donation_data.recipient_id
+
+    if (recipient) {
+      let acct = (await context.db.Account.select({ id: recipient }))[0]
+      console.log("selected acct", acct)
+      let acctUpdate = {
+        total_donations_usd: acct.total_donations_usd || 0 + total_amount_usd,
+        donors_count: acct.donors_count || 0 + 1
+      }
+      await context.db.Account.update({ id: recipient }, acctUpdate)
+    }
+
+    let activity = {
+      signer_id: signerId,
+      receiver_id: receiverId,
+      timestamp: donation.donated_at,
+      type:
+        actionName == "donate"
+          ? "Donate_Direct"
+          : donation.matching_pool
+          ? "Donate_Pot_Matching_Pool"
+            : "Donate_Pot_Public",
+      action_result: recipient,
+      tx_hash: receiptId,
+    };
+
+    await context.db.Activity.insert(activity);
+  }
+
+  // map through the successful actions and swittch on the methodName called for each, then call the designated handlerFunction.
+  await Promise.all(
+    matchingActions.flatMap((action) => {
+      action.operations.map(async (operation) => {
+        console.log("see the contents here...:,==", operation["FunctionCall"]);
+        let call = operation["FunctionCall"];
+        if (call) {
+          const args = atob(call.args); // decode function call argument
+          switch (call.methodName) {
+            case "new":
+            // new can be called on  a couple of things, if the recever id matches factory pattern, then it's a new factory, else if it matches the registry account id, then it's a registry contract initialization, else, it's a new pot initialization.
+              matchVersionPattern(action.receiverId)
+                ? await handleNewFactory(
+                    args,
+                    action.receiverId,
+                    action.signerId,
+                    action.receiptId
+                  )
+                : action.receiverId == "registry.potlock.near" // initializing registry
+                ? await handleRegistry(
+                    args,
+                    action.receiverId,
+                    action.signerId,
+                    action.receiptId
+                  )
+                : await handleNewPot(
+                    args,
+                    action.receiverId,
+                    action.signerId,
+                    action.predecessorId,
+                    action.receiptId
+                  );
```

Comment:
careful of this, because there are other subaccounts of `potlock.near` that are not Pots. Better to call `handleNewPot` based on whether the receiverId is a subaccount of a potfactory account.